cmd/commands/git-source.go

```diff
@@ -294,7 +301,139 @@ func createCronExamplePipeline(opts *gitSourceCronExampleOptions) error {
 		return fmt.Errorf("failed to write yaml of cron example sensor. Error: %w", err)
 	}
 
+	err = cleanUpRedundantFields(sensorFilePath, opts.gsCloneOpts.FS, sensor)
+	if err != nil {
+		return fmt.Errorf("failed to clean up redundant fields of eventSource. Error: %w", err)
+	}
+
+	return nil
+}
+
+func cleanUpRedundantFields(filename string, gsFs fs.FS, o ...interface{}) error {
```

Comment:
In general the logic LGTM but it seems like this function tries to remove all possible redundant fields for all types of resources which seems a bit complicated and can grow in size overtime. Also, this way, we are performing lots of unnecessary operations when for example we are sending here a cron eventsource and trying to remove fields which will only be present on a github eventsource. Maybe we could split this function into multiple functions like `cleanUpCronEventsource`, `cleanUpGithubEventsource`, etc, and have an additional function that can be reused for deleting common redundant fields? WDYT?